kubectl-cloudflow/version/version.go

```diff
@@ -30,21 +33,58 @@ const ProtocolVersionKey = "protocol-version"
 // ProtocolVersionConfigMapName is the name of the configmap that contains the protocol-version
 const ProtocolVersionConfigMapName = "cloudflow-protocol-version"
 
-// FailOnProtocolVersionMismatch fails and exits if the protocol version of kubectl-cloudflow does not match with the cloudflow operator protocol version.
-func FailOnProtocolVersionMismatch() {
+// CloudflowNamespaceLabelKey is the key of the label for the cloudflow installation namespace
+const cloudflowNamespaceLabelKey = "namespace"
+
+// CloudflowNamespaceLabelValue is the value of the label for the cloudflow installation namespace
+const cloudflowNamespaceLabelValue = "cloudflow-namespace"
+
+// GetProtocolVersionConfigMap Get the protocol version config map set by the operator
+func GetProtocolVersionConfigMap() (*corev1.ConfigMap, error) {
 	k8sClient, k8sErr := k8s.GetClient()
 	if k8sErr != nil {
 		util.LogAndExit("Failed to create new kubernetes client, %s", k8sErr.Error())
 	}
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{cloudflowNamespaceLabelKey: cloudflowNamespaceLabelValue}}
 
 	var cm *corev1.ConfigMap
-	if configMaps, err := k8sClient.CoreV1().ConfigMaps("").List(metav1.ListOptions{}); err == nil {
-		for i, v := range configMaps.Items {
-			if v.Name == ProtocolVersionConfigMapName {
-				cm = &configMaps.Items[i]
-			}
+	configMaps, err := k8sClient.CoreV1().ConfigMaps("").List(
+		metav1.ListOptions{LabelSelector: labels.Set(labelSelector.MatchLabels).String()})
+	if err == nil {
+		if len(configMaps.Items) > 1 {
+			return nil, errors.New("Multiple cloudflow operators detected in namespace .. exiting")
```

Comment:
```suggestion
			return nil, errors.New("Multiple Cloudflow operators detected in the cluster. This is not supported. Exiting...")
```

Reply:
@agemooij 2 observations .. This `return` is from the control flow where we have detected one instance of the operator, but somehow the name doesn't match with the protocol-version configmap name. Possibly this case will not arise unless we inadvertently change the name in one place (operator) and not the other (cli). Secondly `go lint` complains on the formatting of the message as it has capitalization .. 
<img width="981" alt="Screen Shot 2020-01-15 at 8 30 02 PM" src="https://user-images.githubusercontent.com/107231/72444512-2fa39480-37d6-11ea-9fbe-ada94a90e26f.png">

Reply:
This return AFAIk is for the case when we have found multiple ConfigMaps in the cluster with the exact same label key and value, e.g. somehow the cluster has multiple instances of the operator running in different namespaces. I think that is the situation we are then reporting an error about since we do not support such clusters.

Cloudflow is a product name so it should be capitalized. Perhaps it is complaining about the trailing dots? Whenever some message to the console says "...ing" I read it as something that is continuing after I have read it, hence the elipsis at the end. I can live without that one if it causes such issues.

Reply:
yeah .. I think we will never get into this `return` statement since all matches in the label selector and comparisons are done using `ProtocolVersionConfigMapName`. Will remove that `return` ..